### PR TITLE
Change score calculation to be independent of weight

### PR DESCRIPTION
**Pull request by kluo**
@stvstnfrd 

Led to stale progress page data when weight was changed from 0 to non-zero.

### CODE REVIEW

File: freetextresponse/freetextresponse.py [hunk 1]
```diff
@@ -393,18 +393,13 @@ def _compute_score(self):
...
-        else:
-            self.score = 0.0
+        self.score = credit
         self.runtime.publish(
             self,
             'grade',
             {
                 'value': self.score,
-                'max_value': self.weight
+                'max_value': 1.0
```
Comment (stvstnfrd) [inline freetextresponse/freetextresponse.py]: Should we set this to `Credit.full`?

**Comment by stvstnfrd** [comment]
This matches my understanding of the issue.

@dcadams Have additional thoughts?

**Comment by stvstnfrd** [comment]
I'm wondering if we should even keep `weight` in here at this point; 
what purpose does it serve?

I see we're using it for some of the UI, but will that be misleading if it has to stay in-sync with the grading scale set in CMS? Should we instead just say "50%" or "100%" or "0%"?

We may want to check with CourseOps to see what the intent here.

File: freetextresponse/freetextresponse.py [hunk 2]
```diff
@@ -393,18 +393,13 @@ def _compute_score(self):
...
         """
         credit = self._determine_credit()
-        if credit == Credit.full:
-            self.score = self.weight
-        elif credit == Credit.half:
-            self.score = float(self.weight)/2
-        else:
-            self.score = 0.0
+        self.score = credit
         self.runtime.publish(
```
Comment (dcadams) [inline freetextresponse/freetextresponse.py:398]: I think a manifestation of this problem is that "score" in coursware_studentmodule.state has been getting set to weight e.g. '{"student_answer": "What do you expect of me? I have no instructions.", "score": 0, "count_attempts": 1}'
whereas I believe it should be set to the score. This is probably how it is able to now recalculate grade after the weight has been changed. Can you confirm my understanding is correct?

Comment (kluo) [reply freetextresponse/freetextresponse.py:398]: Right

**Comment by kluo** [comment]
@stvstnfrd  Well you still need weight for when you have multiple problems in a subsection, then the weight matters for how much the free text response should count for that sections overall grade

**Comment by stvstnfrd** [comment]
+1

**Pull request merged**